keps/sig-node/3008-cri-class-based-resources/README.md

```diff
@@ -0,0 +1,1257 @@
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+<!--
+This section is incredibly important for producing high-quality, user-focused
+documentation such as release notes or a development roadmap. It should be
+possible to collect this information before implementation begins, in order to
+avoid requiring implementors to split their attention between writing release
+notes and implementing the feature itself. KEP editors and SIG Docs
+should help to ensure that the tone and content of the `Summary` section is
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+
+We would like to add support for class-based resources in Kubernetes.
+Class-based resources can be thought of as non-accountable resources, each of
+which is presented by a set of classes. Being non-accountable means that
+multiple containers can be assigned to the same class. They are also supposed
+to be opaque to the CRI client in the sense that the container runtime takes
+care of configuration and control of the resources and the classes within.
+
+A prime example of a class-based resource is Intel RDT (Resource Director
+Technology). RDT is a technology for controlling the cache lines and memory
+bandwidth available to applications. RDT provides a class-based approach for
+QoS control of these shared resources: all processes in the same hardware class
```

Comment:
is there any assumption about classes precedence order? For example, will users desire the scenario when they tell "at least bronze", but if "golden" available - take this one?

Reply:
does it make sense from the hardware description of classes perspective? Or classes are typically defined in some orthogonal dimensions that cannot be linearly ordered from worst to the best?

Reply:
Another good question. I think in general case you cannot assume any ordering from "worst" to "best". E.g. in the case of RDT you might have two classes (say  hi-A and hi-B) with exactly the same amount of cache lines for both of the classes, but reserved exclusively i.e. the cache lines would not overlap. So they could be equally good but interference between apps in different classes would be minimized (in terms of cache).